dashboards-observability/common/constants/autocomplete.ts

```diff
@@ -64,37 +67,127 @@ export interface DataItem {
   doc_count: any;
 }
 
-const JUST_SOURCE_REGEX = new RegExp('source = (\\S+)');
-const SOURCE_WHERE_REGEX = new RegExp('source = (\\S+) \\| where \\S+ = \\S+');
-const SOURCE_MATCH_REGEX = new RegExp('source = (\\S+) \\| where match\\(\\S+, \\S+\\)');
-export const EMPTY_REGEX = new RegExp('\\s+');
-export const MATCH_FIELD_AFTER_WHERE = new RegExp('\\s*where\\s*\\S*');
-export const EQUAL_AFTER_FIELD = new RegExp('\\s*where\\s*(\\S+)\\s+');
-export const DATA_AFTER_EQUAL = new RegExp('\\s*where\\s*\\S+\\s*=\\s*\\S*');
-export const PIPE_AFTER_WHERE = new RegExp('\\s*where\\s*\\S+\\s*=\\s*\\S+\\s+');
-export const FIELD_AFTER_MATCH = new RegExp('\\s*where\\s*match\\(\\s*\\S*');
-export const COMMA_AFTER_FIELD = new RegExp('\\s*where\\s*match\\(\\s*(\\S+)\\s+');
-export const DATA_AFTER_COMMA = new RegExp('\\s*where\\s*match\\(\\s*\\S+\\s*,\\s*');
-export const CLOSE_AFTER_DATA = new RegExp('\\s*where\\s*match\\(\\s*\\S+\\s*,\\s*\\S+\\s+');
-export const PIPE_AFTER_MATCH = new RegExp(
-  '\\s*where\\s*match\\(\\s*\\S+\\s*,\\s*\\S+\\s*\\S+\\s*\\)\\s*'
+const JUST_SEARCH_REGEX = new RegExp('\\s*search\\s+source\\s*=\\s*(\\S+)');
```

Comment:
are capture groups used? If so could you add some comments on what the regex is capturing? for example this seems to capture the index name, although fails for `source = index1, index2`

also just a note when regex doesn't have variables, `var = /\s*search\s+source\s*=\s*(\S+)/` might be cleaner since backslash doesn't have to be escaped

Reply:
Capture groups are mostly used for getting the index and fields but there are places where I am not using the capture groups but the lack of their presence seemed to confuse the matching.

Reply:
Did not know about the being able to use `/` around regex to not have to escape!! I will definitely do that instead. Gosh I wish I had known, I had to delete/add all the escaping every time I was debugging on regex101

Reply:
I will also be adding regex patterns for matching multiple indices in a later PR